### PR TITLE
Feature/queue authevent

### PR DESCRIPTION
**Pull request by acn-dgopa**
## Description

- added queue service that queues the authentication event logging
- added an entry for logging in authentication controller

## Related Issue(s)
- https://github.com/Altinn/altinn-auth-audit-log/issues/19

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Review by TheTechArch — APPROVED** [review]
Looks Good. The FeatureManagement package was new for me. 
Will be existing to see the first logs in database

File: src/Authentication/appsettings.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "MaskinportenWellKnownAlternativeConfigEndpoint": "https://test.maskinporten.no/.well-known/oauth-authorization-server",
     "IdPortenWellKnownConfigEndpoint": "https://oidc-ver2.difi.no/idporten-oidc-provider/.well-known/openid-configuration",
     "IdPortenAlternativeWellKnownConfigEndpoint": "https://oidc-test.difi.no/idporten-oidc-provider/.well-known/openid-configuration",
-    "OpenIdWellKnownEndpoint": "http://localhost:5040/authentication/api/v1/openid",
+    "OpenIdWellKnownEndpoint": "http://localhost:5040/authentication/api/v1/openid/.well-known/openid-configuration",
```
Comment (jonkjetiloye) [inline src/Authentication/appsettings.json:19]: Was this a neccessary change for dev-only?

Based on the values in values.yaml for authentication which sets this config value for the environments it's the issuer not actually the well-known endpoint.

File: src/Authentication/appsettings.json [hunk 2]
```diff
@@ -29,5 +29,14 @@
     "CertificatePath": "jwtselfsignedcert.pfx"
   },
   "OidcProviders": {
+  },
+  "QueueStorageSettings": {
```
Comment (jonkjetiloye) [inline src/Authentication/appsettings.json:33]: Have the QueueStorageSettings been added to keyvault for deploy to the different environments?

File: src/Authentication/appsettings.json [hunk 3]
```diff
@@ -29,5 +29,14 @@
...
+    "ConnectionString": "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;QueueEndpoint=http://127.0.0.1:10001/devstoreaccount1",
+    "QueueEndpoint": "http://127.0.0.1:10001/devstoreaccount1",
+    "AccountKey": "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==",
+    "AuthenticationEventQueueName": "eventlog"
+  },
+  "FeatureManagement": {
+    "AuditLog": true
```
Comment (jonkjetiloye) [inline src/Authentication/appsettings.json:40]: Should this be true by default in appsettings.json? or set pr. environment through values.yaml?

File: src/Authentication/Model/AuthenticationEventType.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Altinn.Platform.Authentication.Model
+{
+    /// <summary>
+    /// Enumeration for authentication event types
+    /// </summary>
+    public enum AuthenticationEventType
+    {
+        Authenticated,
```
Comment (jonkjetiloye) [inline src/Authentication/Model/AuthenticationEventType.cs:8]: Enum should be declared with values and 0 should be NotAuthenticated. Now the default will be Authenticated

**Review by jonkjetiloye — COMMENTED** [review]
Added some comments for consideration.
Have the bug and code smells been looked at?

File: src/Authentication/appsettings.json (hunk 1 above)
Comment (acn-dgopa) [reply src/Authentication/appsettings.json:19]: ya right, I had to change that for dev only. I later moved all my changes to appsettings.Development.json for development. I will revert this

File: src/Authentication/appsettings.json (hunk 2 above)
Comment (acn-dgopa) [reply src/Authentication/appsettings.json:33]: No, I'm yet to create the values.yaml PR

File: src/Authentication/appsettings.json (hunk 3 above)
Comment (acn-dgopa) [reply src/Authentication/appsettings.json:40]: Good question. I think i will set this to false by default and turn on in a specific environment where i can test it out

File: test/Altinn.Platform.Authentication.Tests/Services/EventLogServiceTest.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+using Altinn.Platform.Authentication.Tests.Mocks;
+using Azure.Messaging;
+using Moq;
+using Xunit;
+
+namespace Altinn.Platform.Authentication.Tests.Services
+{
+    public class EventLogServiceTest
+    {
+        //public EventLogServiceTest() { }
```
Comment (jonkjetiloye) [inline test/Altinn.Platform.Authentication.Tests/Services/EventLogServiceTest.cs:20]: Out commented code

**Pull request merged**